include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -146,7 +146,40 @@ def VariableOp : MooreOp<"variable", [
   let arguments = (ins StrAttr:$name, Optional<UnpackedType>:$initial);
   let results = (outs Res<UnpackedType, "", [MemAlloc]>:$result);
   let assemblyFormat = [{
-    `` custom<ImplicitSSAName>($name) ($initial^)? attr-dict
+    custom<ImplicitSSAName>($name) ($initial^)? attr-dict
+    `:` type($result)
+  }];
+}
+
+def NetOp : MooreOp<"net", [
```

Comment:
Since you are adding a new dialect operation, please also add a parsing/printing test to `test/Dialect/Moore/basic.mlir`, ideally one test for each possible form of the declaration. With and without name. With and without the optional assignment. And then the twelve net types. `VarOp` should already have a similar test, but without the net type.